pypeit/scripts/tellfit.py

```diff
@@ -0,0 +1,186 @@
+                        "file, make sure to set the pca_lower and pca_upper in the tell_file to specify the \n"
+                        "wavelength coverage of your model. The defaults are pca_lower=1200. and pca_upper=3100.")
+    parser.add_argument("-t", "--tell_file", type=str, help="R|Configuration file to change default telluric parameters.\n"
+                        "Note that the parameters in this file will be overwritten if you set argument in your terminal. \n"
+                        "The --tell_file option requires a .tell file with the following format:\n"
+                        "\n"
+                        "    [tellfit]\n"
+                        "         objmodel = qso\n"
+                        "         redshift = 7.6\n"
+                        "         bal_mask = 10825,12060\n"
+                        "OR\n"
+                        "    [tellfit]\n"
+                        "         objmodel = star\n"
+                        "         star_type = A0\n"
+                        "         star_mag = 8.\n"
+                        "OR\n"
+                        "    [tellfit]\n"
+                        "         objmodel = poly\n"
+                        "         polyorder = 3\n"
+                        "         fit_region_mask = 9000.,9500.\n"
+                        "\n"
+                        )
+    parser.add_argument("-r", "--redshift", type=float, default=None, help="Object redshift")
+    parser.add_argument("--debug", default=False, action="store_true", help="show debug plots?")
+    parser.add_argument("--plot", default=False, action="store_true", help="Show the telluric corrected spectrum")
+    parser.add_argument("--par_outfile", default='telluric.par', help="Name of outut file to save the parameters used by the fit")
+
+    if options is None:
+        args = parser.parse_args()
+    else:
+        args = parser.parse_args(options)
+    return args
+
+
+def main(args):
+    """
+    Executes telluric correction.
+    """
+
+    # Determine the spectrograph
+    header = fits.getheader(args.spec1dfile)
+    spectrograph = load_spectrograph(header['PYP_SPEC'])
+    spectrograph_def_par = spectrograph.default_pypeit_par()
+
+    # If the .tell file was passed in read it and overwrite default parameters
+    if args.tell_file is not None:
+        cfg_lines = read_tellfile(args.tell_file)
+        par = pypeitpar.PypeItPar.from_cfg_lines(cfg_lines=spectrograph_def_par.to_config(),
+                                                 merge_with=cfg_lines)
+    else:
+        par = spectrograph_def_par
+
+    # If args was provided override defaults. Note this does undo .tell file
+    if args.objmodel is not None:
+        par['tellfit']['objmodel'] = args.objmodel
+    if args.pca_file is not None:
+        par['tellfit']['pca_file'] = args.pca_file
+    if args.redshift is not None:
+        par['tellfit']['redshift'] = args.redshift
+
+    if args.tell_grid is not None:
```

Comment:
So far we have been making the telluric grid an attribute in the spectrograph file. So I'm not sure you need this argument?

Reply:
done